Sources/DolbyIORTSCore/State/StateMachine.swift

```diff
@@ -111,35 +90,26 @@ final actor StateMachine {
 
     func onSubscribed() {
         switch currentState {
-        case .subscribing:
-            currentState = .subscribed(.init())
+        case .subscribed:
+            // No-op
+            break
         default:
-            Self.logger.error("🛑 Unexpected state on onSubscribed - \(self.currentState.description)")
+            currentState = .subscribed(.init())
         }
     }
 
     func onSubscribedError(_ reason: String) {
-        switch currentState {
-        case .connected, .connecting, .subscribed, .subscribing, .error:
-            currentState = .error(.init(error: .subscribeFailed(reason: reason)))
-
-        default:
-            Self.logger.error("🛑 Unexpected state on onSubscribedError - \(self.currentState.description)")
-        }
+        currentState = .error(.init(error: .subscribeFailed(reason: reason)))
     }
 
     func onSignalingError(_ message: String) {
-        switch currentState {
-        case .connected, .connecting, .subscribed, .subscribing:
-            currentState = .error(.init(error: .signalingError(reason: message)))
-
-        default:
-            Self.logger.error("🛑 Unexpected state on onSignalingError - \(self.currentState.description)")
-        }
+        currentState = .error(.init(error: .signalingError(reason: message)))
     }
 
     func onActive(_ streamId: String, tracks: [String], sourceId: String?) {
         switch currentState {
+        case .stopped:
+            currentState = .subscribed(.init())
```

Comment:
When we originally wrote this StateMachine in previous SDK version, the order of callback was `onSubscribed` followed by `onActive` events and in an `onActive` event we start adding source informations one by one.

It would be good to check with Yousif if this implementation has changed before we doing this change ?

Also I notice here - we are setting the state to `subscribed` but not adding the source information, won't it result in that specific source information be lost ?

Reply:
There's a workaround in onInactive() which sets the state to `.stopped` if the sources are empty. After that if we receive `onActive` the .stopped state should be considered as valid (otherwise the stateMachine with stuck).

Reply:
Is that the SDK auto connecting where we get `onActive` even without a `onSubsctibed` callback - we might need to disabled the auto reconnect option in SDK ?

Reply:
Actually looks like you're right, this change can sometimes add blinking in multiview. Thanks a lot, removed it..